pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -348,6 +349,23 @@ public static void validateIngestionConfig(TableConfig tableConfig, @Nullable Sc
           }
         }
       }
+
+      // Complex configs
+      ComplexTypeConfig complexTypeConfig = ingestionConfig.getComplexTypeConfig();
+      if (complexTypeConfig != null && schema != null) {
+        Map<String, String> prefixesToRename = complexTypeConfig.getPrefixesToRename();
+        Set<String> fieldNames = schema.getFieldSpecMap().keySet();
+        if (prefixesToRename != null) {
+          for (String prefix : prefixesToRename.keySet()) {
+            for (String field : fieldNames) {
+              if (field.startsWith(prefix)) {
```

Comment:
(minor) Use `Preconditions.checkState()` with exception message template to simplify this for readability